kbase/narrative
# PTV-1810 - spruce up select2objectview, add tests

# Description of PR purpose/changes

The short version of the PTV-1810 ticket is that when object names change or when their version gets bumped, they get lost in existing apps. The overall fix is not to store the object name in the metadata, but to store the ref. However, this means refactoring both Select2ObjectView and Select2ObjectInput to handle loading/storing refs and some other things.

This PR updates the View form of the widget that appears in the "View Configure" tab of app cells. We do a few things here:
1. accept either strings (i.e. object names) or UPAs as inputs
2. if the input is an UPA (or a workspace_name/object_name ref), look up the relevant object info and display the name from it.
3. if not, just show the given string.

It also fixes a few other small things.
1. This uses code that had a side effect of setting up a data panel listener that wasn't always reaped.
2. This had the Select2 code for interactive usage, when the dropdown is non-interactive.
3. Removes a number of other code pieces to support interactivity that isn't applicable.

The next PR will address a couple issues with the Select2ObjectInput widget.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/PTV-1810

# Testing Instructions
* Details for how to test the PR:
1. Open an existing narrative with an app cell and go to the View Configure tab - it should work as it used to.
2. Run an app cell (on CI only), this should store the UPA right now, and the View Configure tab should display the name properly (with object version number)

# Dev Checklist:

- [n/a] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook

# Updating Version and Release Notes (if applicable)

- [n/a] [Version has been bumped](https://semver.org/) for each release
